New model docs

DarshitChanpura opened the pull request:
Updates/adds the documentation for the models.

Commit:
update: docs for new models

Commit:
update: gemfile update

Commit:
update: deleted model-docs for models that weren`t touched in db-restructure

NdabaNdebele commented on Gemfile:
```diff
@@ -1,11 +1,14 @@
+gem 'middleman', '>= 4.2.1'
+gem 'middleman-syntax', '>= 3.0.0'
+gem 'middleman-autoprefixer', '>= 2.7.0'
+gem "middleman-sprockets", ">= 4.1.0"
```
why is your config different here? I don't think we should commit this

DarshitChanpura replied:
I wasn't sure..so waited on your comment for this..removed it now

NdabaNdebele commented on source/includes/api/_segments.md:
````diff
@@ -1,489 +0,0 @@
-## The segment object
-
-```json
-{
````
why did you delete this?

DarshitChanpura replied:
shouldn't have...a mistake on my end

NdabaNdebele commented on source/includes/api/_organizations.md:
```diff
@@ -1,180 +1,183 @@
 
 ### meta
 
-[See documentation](#metadata-object).
```
nothing was changed in organization, so don't commit these changes

DarshitChanpura replied:
we removed `permissions` field from campaigns schema in endgame branch

NdabaNdebele commented on source/includes/api/_campaigns.md:
```diff
@@ -1,631 +1,872 @@
+					"path": "/data/profile/name/anglican/given",
+					"type": "field"
+				},
+				{
```
remove the changes for this

DarshitChanpura replied:
done

Commit:
fix: restored deleted files and reverted changes to campaigns markdown

Commit:
fix: reverted changes in Gemfile

NdabaNdebele commented on source/includes/api/_campaigns.md:
````diff
@@ -132,7 +132,7 @@ value | true | String value used to separate other fields.
 ```
 http
-POST /campaigns HTTP/1.1
+POST /v1/campaigns HTTP/1.1
````
don't commit any changes to campaigns

NdabaNdebele commented on source/includes/api/_contacts.md:
```diff
@@ -1,677 +1,852 @@
 --------- | --------- | -----------
-listId | true | List identifier this contact belongs to.
-profile | true | Contact profile [details](#data-profile).
+profile | true | Contact profile
```
ok look at profile and see the link to the profiles section, can you add this? the best way to do this is just to manually change the keys on the original contacts document

DarshitChanpura replied:
Okay ..on it

Commit:
update: added local references inside markdowns

Commit:
fix: minor edits

Commit:
update: moved new docs to v3 folder

Commit:
update: restored v1-layer markdowns

DarshitChanpura committed:
Update index.html.md

DarshitChanpura committed:
Rename _status-groups.md to _statusGroups.md

Commit:
update: added ntoes markdown to v1 documentation

Commit:
Merge branch 'new-model-docs' of https://github.com/PolisPolitics/api-docs-website into new-model-docs

adichills committed:
corrected headers and urls

adichills committed:
changed api.polisapp.com to api.beta.polisapp.com and added info about new bulk endpoints

adichills committed:
Some more edits on transitioning from v1 bulk insert to v2 bulk insert

marcosbozza committed:
Pointing index to v3 api docs

adichills committed:
Added a new line to render the example correctly

adichills committed:
Changes in PR comments - added full url in households  endpoint and fixed notes endpoints

adichills committed:
Added before install to yaml file to fix travis build

adichills committed:
Merge pull request #5 from PolisPolitics/correction-new-model-docs

Api docs correction

hannbarton committed:
added note

hannbarton committed:
Merge pull request #8 from PolisPolitics/bug/query

Updated documentation

adichills committed:
Add notes about sections of endpoint which are currently buggy

adichills committed:
Merge pull request #9 from PolisPolitics/note-bugs-in-endpoint

Add notes about sections of endpoint which are currently buggy

adichills committed:
Adding missing filters description in endpoints

Few of the endpoints such as lists , households, interactions , memberships and organizations were missing description for allowed filters . And the note about endpoints throwing errors have been removed as they have been fixed.

adichills committed:
Removed note section on contacts

adichills committed:
Merge pull request #10 from PolisPolitics/correct-allowed-filters

Adding missing filters description in endpoints

adichills committed:
Minor correction in the notes endpoint documentation

Seperated out the the request and response sections in get notes endpoint

adichills committed:
Merge pull request #11 from PolisPolitics/minor-correction-notes-endpoint

Minor correction in the notes endpoint documentation

adichills committed:
Minor formatting changes

adichills committed:
Merge pull request #12 from PolisPolitics/minor-correction-notes-endpoint

Minor formatting changes

adichills committed:
fixed typo on patch of interaction

- Fixed the typo in the example for patch of interaction.
- Added description for turfId in the data of interaction.

adichills committed:
Merge pull request #13 from PolisPolitics/bug/patch-interactions

fixed typo on patch of interaction

gmocellin committed:
Removed Create and Delete doc for segments and added a note

gmocellin committed:
Removed notes

adichills committed:
Merge pull request #14 from PolisPolitics/bug/segment-doc-fix

Removed Create and Delete doc for segments and added a note

Commit:
Removes erroneous slash in endpoint to fetch a contact's households

anjiemerchant committed:
Merge pull request #15 from PolisPolitics/bug/remove-forward-slash-in-endpoint

Removes erroneous slash in endpoint to fetch a contact's households

adichills committed:
Update households endpoint with externalId documentation

adichills committed:
Merge pull request #16 from PolisPolitics/feature/update-household-externalId-documentation

Update households endpoint with externalId documentation

adichills committed:
Update documentation with contact household campaign endpoints

Describe the GET, PATCH and PUT endpoints

adichills committed:
Add documentation to describe the contact household campaign object

adichills committed:
Add section header above meta link in contact household campaign documentation

adichills committed:
Remove redundant paragraph in the Get chc record section

adichills committed:
Address comments on the PR

adichills committed:
Change title of update chc to update status of a chc record

adichills committed:
Merge pull request #17 from PolisPolitics/feature/contact-household-campaign-endpoints

Feature/contact household campaign endpoints

adichills committed:
Update contact endpoints documentation

Add documentation regarding returning the existing contact if there is a conflict and correct some examples

anjiemerchant committed:
Merge pull request #18 from PolisPolitics/feature/update-contacts-endpoint

Update contact endpoints documentation

adichills committed:
Add better wording in the note section in create contacts

adichills committed:
Add some highlighting in the note section of create contacts

adichills committed:
Fix spacing in period

anjiemerchant committed:
Merge pull request #19 from PolisPolitics/feature/update-contacts-endpoint

Add better wording in the note section in create contacts

Commit:
Updates CHC statusId information

anjiemerchant committed:
Merge pull request #20 from PolisPolitics/add-statusId-documentation

Update CHC statusId information

adichills committed:
Update documentation with bulk upsert chc endpoint

adichills committed:
Fix typo in upsert chc bulk endpoint

adichills committed:
Add missing article in the chc bulk upsert enddpoint description

adichills committed:
Merge pull request #21 from PolisPolitics/feature/bulk-upsert-contact-household-campaign

Update documentation with bulk upsert chc endpoint

adichills committed:
Fix documentation errors

- fix missing content type in contact-household-campaign endpoint examples
- fix the method name and url in bulk upsert contact-household-campaigns endpoint

adichills committed:
Merge pull request #22 from PolisPolitics/documentation-errors

Fix documentation errors

gmocellin committed:
Update campaign misspell

gmocellin committed:
Merge pull request #23 from PolisPolitics/bug/fix-campaign-misspell

Update campaign misspell

Commit:
Fix country documentation and displayed values

luizrcb committed:
Merge pull request #24 from PolisPolitics/bug/country-documentation-and-values

Fix country documentation and displayed values
